astropy/io/ascii/ui.py

```diff
@@ -161,10 +162,13 @@ def get_reader(Reader=None, Inputter=None, Outputter=None, **kwargs):
     reader : `~astropy.io.ascii.BaseReader` subclass
         ASCII format reader instance
     """
-    # This function is a light wrapper around core._get_reader to provide a public interface
-    # with a default Reader.
+    # This function is a light wrapper around core._get_reader to provide a
+    # public interface with a default Reader.
     if Reader is None:
-        Reader = basic.Basic
+        if kwargs.get('fast_reader', False) and len(kwargs.get('delimiter', ' ')) < 2:
```

Comment:
IIRC the core problem that guessing would start with a non-fast reader even if `fast_reader` was explicitly requested. Should probably try to update this to test for `kwargs['fast_reader']['enable']` now.

Reply:
Yes, using `kwargs['fast_reader']['enable']` would be more appropriate.  The test on `delimiter` seems a bit ad-hoc and not necessarily robust.  

What would also be very useful is documenting a case where the current code is not doing the best thing.  Even better is creating a test that fails in 3.0 and which is fixed by this update.  E.g. you can look at the last entry in the read trace and confirm that the input was actually read by a fast reader.

Reply:
The part testing `len(delimiter)` would still stay in to avoid calling the fast reader with unsupported multi-char delimiters.
But testing explicitly for the fast reader in the read trace is a good idea; I was not aware of its availability when I first tracked this down (might have saved a number of experiments...)